Problem 1 - Partial Application/partial.js

```diff
@@ -0,0 +1,26 @@
+var iTechArt = iTechArt || {};
+iTechArt.training = iTechArt.training || {};
+iTechArt.training.utils = iTechArt.training.utils || {};
+
+
+/**
+ * Implement function F that allows to do partial function application in form of:
+ *
+ *   G(x, y, z...) = N
+ *
+ *   F(x, G(x, y, z...)) -> H(y, z) = N
+ *
+ * F should accept any number of parameters to apply.
+ * G may accept any number of parameters.
+ */
+
+iTechArt.training.utils.partial = function(fn /* fn -> G function */){
```

Comment:
A minor hint: making function an implicit parameter (i.e. writing partial as : Function.prototype.partial = ... )
will help you to avoid compromises related to passing arguments to the method